src/util/request-builder.js

```diff
@@ -60,32 +91,83 @@ export default class RequestBuilder {
    * @returns {Promise.<any>} Service response as promise
    */
   doHttpRequest(): Promise<any> {
+    const promise = new Promise((resolve, reject) => {
+      this._requestPromise = {
+        resolve,
+        reject
+      };
+    });
     if (!this.url) {
-      throw new Error('serviceUrl is mandatory for request builder');
+      this._requestPromise.reject(
+        new Error(Error.Severity.CRITICAL, Error.Category.NETWORK, Error.Code.MALFORMED_DATA_URI, {
+          url: this.url
+        })
+      );
     }
+    this._createXHR();
+    return promise;
+  }
+
+  _createXHR(): void {
     let request = new XMLHttpRequest();
-    return new Promise((resolve, reject) => {
-      request.onreadystatechange = function() {
-        if (request.readyState === 4) {
+    request.onreadystatechange = () => {
+      if (request.readyState === 4) {
+        try {
           if (request.status === 200) {
-            let jsonResponse;
-            try {
-              jsonResponse = JSON.parse(request.responseText);
-            } catch (e) {
-              return reject(`${e.message}, ${request.responseText}`);
-            }
-            if (jsonResponse && typeof jsonResponse === 'object' && jsonResponse.code && jsonResponse.message) reject(jsonResponse);
-            else resolve(jsonResponse);
+            const response = JSON.parse(request.responseText);
+            this.responseHeaders = this._getResponseHeaders(request);
```

Comment:
add comment to explain why we can't return the answer with the headers here - backward compatibility